File: SEARCH/linearSearch/linearSearch.java
public class linearSearch {
    
    // Метод осуществляет линейный поиск элемента target в массиве arr
    public static int linearSearch(int[] arr, int target) {        
        for (int i = 0; i < arr.length; i++) {      // Проходим по каждому элементу массива
            if (arr[i] == target) {                // Если нашли искомый элемент...
                return i;                          // Возвращаем индекс найденного элемента
            }
        }
        return -1;                                 // Если элемент не найден, возвращаем -1
    }

    public static void main(String[] args) {
        int[] array = {10, 20, 80, 30, 60, 50};   // Создаем тестовый массив

        int keyToFind = 80;                        // Элемент, который мы хотим найти

        int resultIndex = linearSearch(array, keyToFind); // Запускаем метод поиска

        if(resultIndex != -1){
            System.out.println("Element found at index: " + resultIndex);
        } else{
            System.out.println("Element not found in the array");
        }
    }
}

File: lab6/SEARCH/linearSearch/linearSearch.py
def linear_search(arr, target):
    # Перебор всех элементов списка arr
    for idx in range(len(arr)):
        # Проверка текущего элемента на равенство искомому значению
        if arr[idx] == target:
            # Возврат индекса первого совпадающего элемента
            return idx
    # Если элемент не найден, возврат -1
    return -1


# Тестируем функцию
array = [10, 20, 80, 30, 60, 50]
key_to_find = 80

# Вызываем функцию поиска
result_index = linear_search(array, key_to_find)

# Анализ результата
if result_index != -1:
    print(f'Элемент найден на индексе: {result_index}')
else:
    print('Элемент не найден')